doc: add note about endpoint.name==service.name (#33197)
kubernetes/website: content/en/docs/concepts/services-networking/service.md

The name of the Endpoints object must be a valid
[DNS subdomain name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).

> {{< note >}} The name of the Endpoint must be the same as the name of the Service. {{< /note >}}

Reviewer: - Kubernetes doesn't have an API type “Endpoint”; did you mean “[Endpoints](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/endpoints-v1/)”
- how does this requirement apply to the [EndpointSlice](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/endpoint-slice-v1/) API?

How about:
```suggestion
When you create an [Endpoints](docs/reference/kubernetes-api/service-resources/endpoints-v1/)
object for a Service, you set the name of the new Endpoints to be the same as the
name of the Service.
```

Reviewer: You could also put a comment into the example Endpoints manifest @Shabirmean

Author: Thank you for the review @sftim. I've adopted the changes. Also I think EndpointSlices don't have this requirement. The association is done via adding an annotation.

The name of the Endpoints object must be a valid
[DNS subdomain name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).

When you create an [Endpoints](docs/reference/kubernetes-api/service-resources/endpoints-v1/)

> object for a Service, you set the name of the new Endpoints to be the same as the

Reviewer 2: I would say "Endpoints object" in every mention of the object. Otherwise it reads a bit strangely. Nit, really.

Author: Done in https://github.com/kubernetes/website/pull/33197/commits/65dbc312c73cbe7873dd63aac60d9c2d0198052c

The name of the Endpoints object must be a valid
[DNS subdomain name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).

> When you create an [Endpoints](docs/reference/kubernetes-api/service-resources/endpoints-v1/)

Reviewer 3: ```suggestion
When you create an [Endpoint](docs/reference/kubernetes-api/service-resources/endpoints-v1/)
```

Author: Avoiding suggestion as per comment https://github.com/kubernetes/website/pull/33197#issuecomment-1113104092

> object for a Service, you set the name of the new Endpoints object to be the same as the

Reviewer 3: ```suggestion
object for a Service, you set the name of the new object to be the same as that
```

> name of the Service.

Reviewer 3: ```suggestion
of the Service.
```

Reviewer: “the same as the of the Service” doesn't look idiomatic to me @divya-mohan0209 - is that definitely the change you wanted to suggest?

Reviewer 3: I meant to suggest "the same as that of the Service".
